src/variorum/config_architecture.h

```diff
@@ -77,7 +77,9 @@ enum nvidia_gpu_arch_e
 
 enum arm_arch_e
 {
-    ARMV8 = 1, //ARM Juno
+    ARM_CORTEX_A72  = 0xd08, //ARM Cortex-A72 MPCore processor
+    ARM_CORTEX_A53  = 0xd03, //ARM Cortex-A53 MPCore processor
```

Comment:
Why are these separate now, shouldn't we be representing the big.Little Juno r2 device as a single entity (we did this in the past)?

Reply:
I can change the model check in `config_arm.c` to what's outlined below to make it clear that we're checking for both A53 and A72 CPUs in the combined big.LITTLE SoC, and not for either one of the ARM CPUs. Does that work?  

```
if (*g_platform[idx].arch_id == ARM_CORTEX_A72 &&
        *g_platform[idx].arch_id == ARM_CORTEX_A53)
```

Reply:
Yes, I think this would address my concern about checking it as a single entity. Thanks @amarathe84 !

Reply:
I changed the `detect_arm_arch()` function to check the CPU ID of big and LITTLE clusters using the `sysfs` interface for `midr_el` register. The fix works for both Neoverse N1 and Juno r2. I did notice that the file I/O has slowed down the architecture check but that's the only way to simplify the logic. All tests worked as expected.